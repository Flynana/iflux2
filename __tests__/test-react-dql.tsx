import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Map } from 'immutable'

import {
  Actor,
  Store,
  StoreProvider,
  Relax,
  DQL,
} from '../src/index'

jest.mock('react-dom')

type IMap = Map<string, any>;

class ProductActor extends Actor {
  defaultState() {
    return {
      products: [
        { id: 1, name: 'p1' },
        { id: 2, name: 'p2' },
        { id: 3, name: 'p3' },
        { id: 4, name: 'p4' },
        { id: 5, name: 'p5' },
        { id: 6, name: 'p6' },
      ]
    }
  }
}

class Appstore extends Store {
  bindActor() {
    return [new ProductActor]
  }
}

@StoreProvider(Appstore)
class ProductApp extends React.Component {
  props: {
    store: Appstore;
  };

  render() {
    const products = this.props.store.state().get('products')

    return (
      <div>
        {products.map((p, index) => <ProductItem index={index} key={p.get('id')} />)}
      </div>
    )
  }
}

const productDQL = DQL('productDQL', [
  ['products', '$index'],
  p => p
])

@Relax
class ProductItem extends React.Component {
  props: {
    product: IMap
  };

  static defaultProps = {
    index: 0,
    product: productDQL
  };

  render() {
    const { id, name } = this.props.product.toJS()

    return (
      <div>
        <div>{id}</div>
        <div>{name}</div>
      </div>
    )
  }
}


describe('react-dql test suite', () => {
  it('initial dql', () => {
    const tree = renderer.create(<ProductApp />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})