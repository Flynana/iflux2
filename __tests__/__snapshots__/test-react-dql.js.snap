exports[`react-dql test suite initial dql 1`] = `
<div>
  <div>
    <div>
      1
    </div>
    <div>
      p1
    </div>
  </div>
  <div>
    <div>
      1
    </div>
    <div>
      p1
    </div>
  </div>
  <div>
    <div>
      1
    </div>
    <div>
      p1
    </div>
  </div>
  <div>
    <div>
      1
    </div>
    <div>
      p1
    </div>
  </div>
  <div>
    <div>
      1
    </div>
    <div>
      p1
    </div>
  </div>
  <div>
    <div>
      1
    </div>
    <div>
      p1
    </div>
  </div>
</div>
`;
